pandas/tests/series/test_combine_concat.py

```diff
@@ -116,8 +116,37 @@ def test_update(self):
         df = DataFrame([{"a": 1}, {"a": 3, "b": 2}])
         df['c'] = np.nan
 
-        # this will fail as long as series is a sub-class of ndarray
-        # df['c'].update(Series(['foo'],index=[0])) #####
+        df['c'].update(Series(['foo'], index=[0]))
+        expected = DataFrame([[1, np.nan, 'foo'], [3, 2., np.nan]],
+                             columns=['a', 'b', 'c'])
+        assert_frame_equal(df, expected)
+
+    def test_update_dtypes(self):
+        s = Series([1., 2., False, True])
+        other = Series([45], index=[0])
```

Comment:
can you use the same variable names in the first part as you do in the last. 

t_src is pretty opaque, just cal lit s and use s_copy

Reply:
I used different names because `s` and `t` are different. Can switch to `t_copy` if you prefer?